tests/components/twinkly/test_twinkly.py

```diff
@@ -0,0 +1,224 @@
+    client.is_offline = True
+    entity, device, _ = await _create_entries(hass, client)
+
+    state = hass.states.get(entity.entity_id)
+
+    assert state.name == TEST_NAME_ORIGINAL
+    assert state.state == "unavailable"
+    assert state.attributes["friendly_name"] == TEST_NAME_ORIGINAL
+    assert state.attributes["icon"] == "mdi:string-lights"
+
+    assert entity.original_name == TEST_NAME_ORIGINAL
+    assert entity.original_icon == "mdi:string-lights"
+
+    assert device.name == TEST_NAME_ORIGINAL
+    assert device.model == TEST_MODEL
+    assert device.manufacturer == "LEDWORKS"
+
+
+async def test_turn_on(hass: HomeAssistant):
+    """Test support of the light.turn_on service."""
+    client = ClientMock()
+    client.is_on = False
+    client.brightness = 20
+    entity, _, _ = await _create_entries(hass, client)
+
+    assert hass.states.get(entity.entity_id).state == "off"
+
+    await hass.services.async_call(
+        "light", "turn_on", service_data={"entity_id": entity.entity_id}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity.entity_id)
+
+    assert state.state == "on"
+    assert state.attributes["brightness"] == 51
+
+
+async def test_turn_on_with_brightness(hass: HomeAssistant):
+    """Test support of the light.turn_on service with a brightness parameter."""
+    client = ClientMock()
+    client.is_on = False
+    client.brightness = 20
+    entity, _, _ = await _create_entries(hass, client)
+
+    assert hass.states.get(entity.entity_id).state == "off"
+
+    await hass.services.async_call(
+        "light",
+        "turn_on",
+        service_data={"entity_id": entity.entity_id, "brightness": 255},
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity.entity_id)
+
+    assert state.state == "on"
+    assert state.attributes["brightness"] == 255
+
+
+async def test_turn_off(hass: HomeAssistant):
+    """Test support of the light.turn_off service."""
+    entity, _, _ = await _create_entries(hass)
+
+    assert hass.states.get(entity.entity_id).state == "on"
+
+    await hass.services.async_call(
+        "light", "turn_off", service_data={"entity_id": entity.entity_id}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity.entity_id)
+
+    assert state.state == "off"
+    assert state.attributes["brightness"] == 0
+
+
+async def test_update_name(hass: HomeAssistant):
+    """
+    Validate device's name update behavior.
+
+    Validate that if device name is changed from the Twinkly app,
+    then the name of the entity is updated and it's also persisted,
+    so it can be restored when starting HA while Twinkly is offline.
+    """
+    entity, _, client = await _create_entries(hass)
+
+    updated_config_entry = None
+
+    async def on_update(ha, co):
+        nonlocal updated_config_entry
+        updated_config_entry = co
+
+    hass.config_entries.async_get_entry(entity.unique_id).add_update_listener(on_update)
```

Comment:
Why do we need to attach a listener? We can just check that the existing config entry has been updated with new data via `hass.config_entries` api.